File: app/src/main/java/com/chuongvd/app/signal/ui/common/BaseDataBindingActivity.java
package com.chuongvd.app.signal.ui.common;

import android.arch.lifecycle.AndroidViewModel;
import android.databinding.DataBindingUtil;
import android.databinding.ViewDataBinding;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;

public abstract class BaseDataBindingActivity<B extends ViewDataBinding, V extends AndroidViewModel>
        extends AppCompatActivity {

    protected B mBinding;

    protected V mViewModel;

    public abstract int getContentViewLayoutId();

    public abstract void initData();

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        mBinding = DataBindingUtil.setContentView(this, getContentViewLayoutId());
        initData();
    }
}

File: app/src/main/java/com/chuongvd/app/signal/ui/common/BaseDataBindingActivity.kt
package com.chuongvd.app.signal.ui.common

import android.arch.lifecycle.AndroidViewModel
import android.databinding.DataBindingUtil
import android.databinding.ViewDataBinding
import android.os.Bundle
import android.support.v7.app.AppCompatActivity

abstract class BaseDataBindingActivity<B : ViewDataBinding, V : AndroidViewModel> : AppCompatActivity() {

  protected lateinit var mBinding: B

  protected lateinit var mViewModel: V

  abstract val contentViewLayoutId: Int

  abstract fun initData()

  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    mBinding = DataBindingUtil.setContentView(this, contentViewLayoutId)
    initData()
  }
}
